tests/visualization_tests/test_edf.py

```diff
@@ -1,106 +1,166 @@
     assert np.all((0 <= np_values) & (np_values <= 1))
 
 
+@parametrized_plot_contour
 @pytest.mark.parametrize("direction", ["minimize", "maximize"])
-def test_plot_optimization_history(direction: str) -> None:
+def test_edf_plot_target_label(plot_edf: Callable[..., Any], direction: str) -> None:
     # Test with no studies.
     figure = plot_edf([])
-    assert len(figure.data) == 0
+    save_static_image(figure)
 
     # Test with no trials.
     figure = plot_edf(create_study(direction=direction))
-    assert len(figure.data) == 0
+    save_static_image(figure)
 
     figure = plot_edf([create_study(direction=direction), create_study(direction=direction)])
-    assert len(figure.data) == 0
+    save_static_image(figure)
 
     # Test with a study.
     study0 = create_study(direction=direction)
     study0.optimize(lambda t: t.suggest_float("x", 0, 5), n_trials=10)
     figure = plot_edf(study0)
-    _validate_edf_values(figure.data[0]["y"])
-    assert len(figure.data) == 1
-    assert figure.layout.xaxis.title.text == "Objective Value"
+    _assert_target_label(figure)
 
     # Test with two studies.
     study1 = create_study(direction=direction)
     study1.optimize(lambda t: t.suggest_float("x", 0, 5), n_trials=10)
     figure = plot_edf([study0, study1])
-    for points in figure.data:
-        _validate_edf_values(points["y"])
-    assert len(figure.data) == 2
-    figure = plot_edf((study0, study1))
-    for points in figure.data:
-        _validate_edf_values(points["y"])
-    assert len(figure.data) == 2
+    _assert_target_label(figure)
 
     # Test with a customized target value.
     study0 = create_study(direction=direction)
     study0.optimize(lambda t: t.suggest_float("x", 0, 5), n_trials=10)
     with pytest.warns(UserWarning):
         figure = plot_edf(study0, target=lambda t: t.params["x"])
-    _validate_edf_values(figure.data[0]["y"])
-    assert len(figure.data) == 1
+    _assert_target_label(figure)
 
     # Test with a customized target name.
     study0 = create_study(direction=direction)
     study0.optimize(lambda t: t.suggest_float("x", 0, 5), n_trials=10)
     figure = plot_edf(study0, target_name="Target Name")
-    _validate_edf_values(figure.data[0]["y"])
-    assert len(figure.data) == 1
-    assert figure.layout.xaxis.title.text == "Target Name"
+    _assert_target_label(figure, expected="Target Name")
+
+
+def test_empty_edf_info() -> None:
+
+    edf_info = _get_edf_info([])
+    assert len(edf_info.lines) == 0
+    assert len(edf_info.x_values) == 0
+
+    study = create_study()
+    edf_info = _get_edf_info(study)
+    assert len(edf_info.lines) == 0
+    assert len(edf_info.x_values) == 0
+
+    trial = study.ask()
+    study.tell(trial, state=optuna.trial.TrialState.PRUNED)
+    edf_info = _get_edf_info(study)
+    assert len(edf_info.lines) == 0
+    assert len(edf_info.x_values) == 0
 
 
 @pytest.mark.parametrize("value", [float("inf"), -float("inf"), float("nan")])
```

Comment:
```suggestion
@pytest.mark.parametrize("value", [float("inf"), -float("inf")])
```
Maybe we should not check `nan` case, because values are never `nan` in optuna.